src/main/java/com/example/bookstore/controller/AuthenticationController.java

```diff
@@ -0,0 +1,25 @@
+package com.example.bookstore.controller;
+
+import com.example.bookstore.dto.user.UserRegistrationRequestDto;
+import com.example.bookstore.dto.user.UserResponseDto;
+import com.example.bookstore.exception.RegistrationException;
+import com.example.bookstore.service.user.UserService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/auth")
+public class AuthenticationController {
+    private final UserService userService;
+
+    @PostMapping("/registration")
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
```

Comment:
make it unchecked, and remove from method definition here and in service. If user submitted incorrect data we can't do anything about it anyway